### DISCUSSION TITLE
Explanation on Evaluation results after performing inference.

### DISCUSSION THREAD

**Discussion by etetteh** [ivadomed/ivadomed#1065] [General]
I ran the [first tutorial](https://ivadomed.org/tutorials/one_class_segmentation_2d_unet.html#train-model) (after multiple attempts to install on CC), but I need a little explanation on the evaluation results after running the testing phase.
Am I right to say that, what is happening there is, random subjects are selected, and the model does a prediction on them?

![Screen Shot 2022-01-31 at 15 45 40](https://user-images.githubusercontent.com/28512232/151872491-f03d9cc5-6f98-467b-86b8-87df9b17d612.JPG)

**Comment by jcohenadad** [comment]
@etetteh since you are 'intern' to the core dev team, I suggest you look at the code that generates this report to figure out what it does, and if needed update the tutorial and/or output display to make it more intuitive to future users.

**Reply by etetteh** [reply]
Okay please. Thank you, and I will do that as soon as possible. Do I need any permission or access to edit the tutorial?

**Reply by jcohenadad** [reply]
👉  https://ivadomed.org/contributing.html

**Comment by uzaymacar** [comment]
Perhaps needless to say, but a comprehensive understanding of the

https://github.com/ivadomed/ivadomed/blob/e6554281d07b5afef9e68ce8b02e86b02bd68363/ivadomed/main.py#L313

function and stepping inside the sub-functions within (e.g. `imed_evaluation` for evaluation) will go a long way!

> Am I right to say that, what is happening there is, random subjects are selected, and the model does a prediction on them?

Yes, you are right. Test subjects are selected based on the `"random_seed"` key in the config file (described [here](https://ivadomed.org/configuration_file.html#random-seed)). Using the same seed across different runs (e.g. `ivadomed --train` and then `ivadomed --test`) ensures the same set of subjects to be selected as test.

**Reply by etetteh** [reply]
Thank you @uzaymacar. I just followed your suggestion, and my doubts has been cleared now.
